Userland/Applications/SpaceAnalyzer/main.cpp

```diff
@@ -183,7 +183,10 @@ ErrorOr<int> serenity_main(Main::Arguments arguments)
 
     auto& file_menu = window->add_menu("&File");
     file_menu.add_action(GUI::Action::create("&Analyze", [&](auto&) {
-        if (auto result = analyze(tree, treemapwidget, statusbar); result.is_error()) {
+        // FIXME: Just modify the tree in memory instead of traversing the entire file system
+        // FIXME: Dispose of the old tree
```

Comment:
Isn't the old tree automatically disposed of when `TreeMapWidget::set_tree()` (inside `analyze()`) assigns the new value to `m_tree`?

Reply:
Well. the reference to the old tree inside the tree map widget is gone, but the original `RefPtr` inside the  `serenity_main` function is never changed/moved/cleaned up

Reply:
Ah, yes. I missed that. Eventually we can stop keeping that RefPtr around I think, but one step at a time.